### PR TITLE
WIP - GitURL with convert to HTTPS

### PR DESCRIPTION
**Pull request by nurali-techie**
Added GitURL with convert to HTTPS functionality.

### CODE REVIEW

File: codebase/git.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+func NewGitURL(url string) (*GitURL, error) {
+	if url == "" {
+		return nil, errors.New("invalid URL, URL is blank")
+	}
+
+	pattern := regexp.MustCompile(`^(https|http|git)(:\/\/|@)([^\/:]+)[\/:]([^\/:]+)\/(.+).git$`)
+	if !pattern.MatchString(url) {
+		return nil, errors.New(fmt.Sprintf("invalid URL, %v", url))
```
Comment (golangci) [inline codebase/git.go]: should use fmt.Errorf(...) instead of errors.New(fmt.Sprintf(...))

File: codebase/git.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+const (
+	HTTPS Scheme = "https"
+)
+
+// There is lot with Git URL - supported scheme, format and validation (regex). Currently, GitURL support very specific conversion.
+// reference:
+// https://mirrors.edge.kernel.org/pub/software/scm/git/docs/git-clone.html#_git_urls_a_id_urls_a
+// https://github.com/git/git/blob/77bd3ea9f54f1584147b594abc04c26ca516d987/urlmatch.c
+// https://stackoverflow.com/questions/2514859/regular-expression-for-git-repository
+// https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols
```
Comment (nurali-techie) [inline codebase/git.go:20]: Parsing any_git_url and Converting to different scheme required lot of implementation and efforts.
Currently,
design - provide complete design to support different scheme conversion in future.
implementation - provided basic implementation to support HTTPS conversion which is needed for WIT codebase search.

File: codebase/git_test.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+		wantErr bool
+	}{
+		{"git@github.com:testuser/testrepo.git", codebase.HTTPS, "https://github.com/testuser/testrepo.git", false},
+		{"https://github.com/testuser/testrepo.git", codebase.HTTPS, "https://github.com/testuser/testrepo.git", false},
+		{"git@github.com:testuser/testrepo", codebase.HTTPS, "", true},
+		{"git@anything/testrepo", codebase.HTTPS, "", true},
+		{"anything/testrepo", codebase.HTTPS, "", true},
+		{"http://github.com/testuser/testrepo.git", codebase.HTTPS, "https://github.com/testuser/testrepo.git", false},
+		{"http://anything/testrepo.git", codebase.HTTPS, "", true},
+	}
```
Comment (nurali-techie) [inline codebase/git_test.go:24]: Added test to check from_scheme support `http`, `git/ssh`.  Also added test with invalid URLs.

**Comment by nurali-techie** [comment]
Check with @aslakknutsen, and decided that No need for GitURL.convert() functionality into f8-common.

**Pull request closed without merging**